x-pack/packages/kbn-elastic-assistant/impl/data_anonymization/settings/anonymization_settings/index.tsx

```diff
@@ -30,50 +30,71 @@ const StatFlexItem = styled(EuiFlexItem)`
 `;
 
 export interface Props {
-  defaultAllow: string[];
-  defaultAllowReplacement: string[];
-  pageSize?: number;
-  setUpdatedDefaultAllow: React.Dispatch<React.SetStateAction<string[]>>;
-  setUpdatedDefaultAllowReplacement: React.Dispatch<React.SetStateAction<string[]>>;
+  defaultPageSize?: number;
+  anonymizationFields: FindAnonymizationFieldsResponse;
+  anonymizationFieldsBulkActions: PerformBulkActionRequestBody;
+  setAnonymizationFieldsBulkActions: React.Dispatch<
+    React.SetStateAction<PerformBulkActionRequestBody>
+  >;
+  setUpdatedAnonymizationData: React.Dispatch<
+    React.SetStateAction<FindAnonymizationFieldsResponse>
+  >;
 }
 
 const AnonymizationSettingsComponent: React.FC<Props> = ({
-  defaultAllow,
-  defaultAllowReplacement,
-  pageSize,
-  setUpdatedDefaultAllow,
-  setUpdatedDefaultAllowReplacement,
+  defaultPageSize,
+  anonymizationFields,
+  anonymizationFieldsBulkActions,
+  setAnonymizationFieldsBulkActions,
+  setUpdatedAnonymizationData,
 }) => {
-  const { baseAllow, baseAllowReplacement } = useAssistantContext();
-
   const onListUpdated = useCallback(
-    (updates: BatchUpdateListItem[]) => {
-      updateDefaults({
-        defaultAllow,
-        defaultAllowReplacement,
-        setDefaultAllow: setUpdatedDefaultAllow,
-        setDefaultAllowReplacement: setUpdatedDefaultAllowReplacement,
-        updates,
+    async (updates: BatchUpdateListItem[]) => {
+      const updatedFieldsKeys = updates.map((u) => u.field);
+
+      const updatedFields = updates.map((u) => ({
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        ...anonymizationFields.data.find((f) => f.field === u.field)!,
```

Comment:
```suggestion
        ...(anonymizationFields.data.find((f) => f.field === u.field) ?? { id: '', field: '' }),
```

This cleared the type errors for me, not sure if having the empty id/field has negative consequences?